internal/driver/virtualdriver.go

```diff
@@ -234,6 +237,12 @@ func prepareVirtualResources(driver *VirtualDriver, deviceName string) error {
 						dr.Properties.Value.Type <-> VIRTUAL_RESOURCE.DataType
 						dr.Properties.Value.DefaultValue <-> VIRTUAL_RESOURCE.Value
 					*/
+					if dsModels.ParseValueType(dr.Properties.Value.Type) == dsModels.Binary {
+						b := make([]byte, dsModels.MaxBinaryBytes)
+						rand.Seed(time.Now().UnixNano())
+						rand.Read(b)
+						dr.Properties.Value.DefaultValue = hex.EncodeToString(b)
```

Comment:
Why are you using dr.Properties.Value.DefaultValue instead of a new variable here?